### PR TITLE
gobjectIntrospection: use absolute path for cairo GIR

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Cairo does not provide its own GObject bindinds so they are provided by gobject-introspection package. Unfortunately, this means that if we want to use the absolute path, we need gi to depend on cairo, which increases the closure size from 41M to 56M. We will probably want to split the typelib into a separate output.

Closes: #34080

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @psychon does this fix your thing?

### CODE REVIEW

**Comment by psychon** [comment]
> cc @psychon does this fix your thing?

I'll try to test tomorrow. (Which means: Getting rid of the `$LD_LIBRARY_PATH` for `awesome` and see if it still works)

**Comment by psychon** [comment]
TL;DR: @jtojnar I can confirm that this patch works.

Running
```env -i HOME=$HOME DISPLAY=:1  \
GI_TYPELIB_PATH='/nix/store/ir2qdvcbj3zbgsicq3cdi2ccschxilw1-pango-1.40.14/lib/girepository-1.0:/nix/store/v8ypi220jcdj3hvw55cjk6b29nin7l1j-gdk-pixbuf-2.36.7/lib/girepository-1.0:/nix/store/3drjms2mf82595cnrlqfikh0x1ifdx5s-gobject-introspection-1.54.1/lib/girepository-1.0:/nix/store/ir2qdvcbj3zbgsicq3cdi2ccschxilw1-pango-1.40.14/lib/girepository-1.0' \
"/nix/store/4s4ca0dja51x7i4r5xvanyqb8k0cvz26-awesome-4.2/bin/.awesome-wrapped"  \
--search /nix/store/21izsmmbpkbpfqlzqzdld9ihg5mz9djb-lgi-0.9.1/share/lua/5.2 \
--search /nix/store/21izsmmbpkbpfqlzqzdld9ihg5mz9djb-lgi-0.9.1/lib/lua/5.2
```
fails with "Failed to load shared library 'libcairo-gobject.so.2' referenced by the typelib: libcairo-gobject.so.2", as expected. The above is just the "normal" in-tree awesome without the `makeWrapper`-wrapper (and with some other unnecessary stuff that the wrapper does also removed).
Then I built the patched `gobjectIntrospection` and added `result/lib/girepository-1.0/:` to the beginning of `GI_TYPELIB_PATH`, so that the new `cairo-1.0.typelib` is used. This resulted in awesome successfully starting.

@jtojnar Thank you for the fix and thank you for coming up with a fix so quickly.
